make.js

```diff
@@ -191,7 +191,8 @@ CLI.build = function()
 CLI.serverBuild = function() {
     ensureBuildTasksAndRemoveTestPath();
 
-    ensureTool('tsc', '--version', 'Version 4.0.8');
+    ensureTool('tsc', '--version', 'Version 4.8.4');
```

Comment:
I'm thinking it would be better to check versions based on the tasks being built to ensure .  E.g. 4.0.2 for most tasks, for one task @JamieMagee updated to 4.0.8 and 4.8.4 for node20?  

For consistency/repeatability, I think it's better if we ensure that the expected typescript version is used for each task

cc: @JamieMagee

Reply:
Also, we should update the code path that uses allowedTypescriptVersions to be strict e.g. demand the specific version of typescript for the build config, rather than allowing any version on the list.  https://github.com/microsoft/azure-pipelines-tasks/blob/87a2ffb621f7b87ff5e5bb9a0f409d7e25230ce3/make-util.js#L167

Related: #18648

Reply:
I removed 4.0.2 and 4.0.8 from the list: `var allowedTypescriptVersions = ['4.8.4'];` 
Not sure if that will impact on any other tasks. This will enforce us to upgrade to 4.8.4 for each task?

Reply:
I wonder if we want to have 4.8.4 for a part of tasks, and for other tasks we keep using 4.0.8.

Reply:
Why 4.8.4?

| TypeScript | Node Version |
| ---------- | ------------ |
| 4.8.4      | >=4.2.0      |
| 4.9.5      | >=4.2.0      |
| 5.0.4      | >=12.20      |
| 5.1.6      | >=14.17      |

I think 4.9.5 makes most sense right now, until every task is upgraded to Node 16.

Reply:
4.8.4 is the least version that node 20 would require on typescript. I'm good with upgrading it to version >= 4.8.4. We can update it to 4.9.5 if that works better.

Reply:
My thinking: 
for node16 (which is hopefully done) keep the tsc version we've been using

for node20 - latest and greatest

The Node20 migration should be a lot more rapid than Node16.  We'll use the build configuration to keep parallel node16 and node 20 builds together so we can easily deploy and test side-by-side.

tagging @kirill-ivlev @tkasparek

Reply:
Okay, so keep TypeScript 4.0.2 for Node 16, and use 5.1.6 for Node 20?

I can revert my 4.0.8 change to keep things simple.

Reply:
Sounds good :)